test/common/router/route_fuzz_test.cc

```diff
@@ -16,6 +18,19 @@ namespace {
 // Limit the size of the input.
 static const size_t MaxInputSize = 64 * 1024;
 
+bool isUnsupportedRouteConfig(const envoy::config::route::v3::Route& route,
+                              bool extended_check = false) {
+  if (route.has_filter_action() || route.has_non_forwarding_action()) {
+    return true;
+  }
+  if (extended_check && route.has_route() &&
+      !(route.route().has_cluster_specifier_plugin() || route.route().has_cluster() ||
+        route.route().has_cluster_header() || route.route().has_weighted_clusters())) {
+    return true;
+  }
```

Comment:
Is this part needed to exclude [`inline_cluster_specifier_plugin `](https://github.com/envoyproxy/envoy/blob/469f7bbd5951a29dfc24a3a05ca01459b835da2c/api/envoy/config/route/v3/route_components.proto#L967)?

IIUC the `cluster_specifier` is a `oneof` that is required, so one of the fields must appear there, no?

Reply:
The `inline_cluster_specifier_plugin` is omitted from the check, because in [`clusterEntry()`](https://github.com/envoyproxy/envoy/blob/911a87020b00e77e7a43fd6924b9579c741261c2/source/common/router/config_impl.cc#L1172-L1189) it is not referenced. Its presence will therefore not suffice to prevent the assert.

Reply:
During research to answer this question, I found another issue, that is not completely covered by the current state. Commit https://github.com/envoyproxy/envoy/pull/21638/commits/dff70d23b4508ab3b2870ae88c987a2cdef1a0df now also validates route action config where the type_url is not exactly given as expected, but only contains `"envoy.config.route.v3.Route"`.